### ISSUE TITLE
How to finetune Flux-dev full params, 80G OOM ...

### ISSUE THREAD

**Issue by huangjun12** [huggingface/diffusers#10196]
I am using the [train_dreambooth_flux](https://github.com/huggingface/diffusers/blob/main/examples/dreambooth/train_dreambooth_flux.py) script to fine-tune the `flux-dev` model with full parameters using DeepSpeed Stage 2. However, I am still encountering out-of-memory issues on an 80GB GPU. Are there any solutions available to address this problem? Thanks!

**Comment by hlky** [comment]
cc @linoytsaban @sayakpaul

**Comment by sayakpaul** [comment]
What is the training command you're using?

**Comment by huangjun12** [comment]
> train_dreambooth_flux

export MODEL_NAME="black-forest-labs/FLUX.1-dev"
export INSTANCE_DIR="dog"
export OUTPUT_DIR="trained-flux"

accelerate launch train_dreambooth_flux.py \
  --pretrained_model_name_or_path=$MODEL_NAME  \
  --instance_data_dir=$INSTANCE_DIR \
  --output_dir=$OUTPUT_DIR \
  --mixed_precision="bf16" \
  --instance_prompt="a photo of sks dog" \
  --resolution=1024 \
  --train_batch_size=1 \
  --guidance_scale=1 \
  --gradient_accumulation_steps=4 \
  --optimizer="prodigy" \
  --learning_rate=1. \
  --report_to="wandb" \
  --lr_scheduler="constant" \
  --lr_warmup_steps=0 \
  --max_train_steps=500 \
  --validation_prompt="A photo of sks dog in a bucket" \
  --validation_epochs=25 \
  --seed="0" \
  --push_to_hub

as in [README_flux.md](https://github.com/huggingface/diffusers/blob/main/examples/dreambooth/README_flux.md)

**Comment by huangjun12** [comment]
> What is the training command you're using?

> train_dreambooth_flux

```
export MODEL_NAME="black-forest-labs/FLUX.1-dev"
export INSTANCE_DIR="dog"
export OUTPUT_DIR="trained-flux"

accelerate launch train_dreambooth_flux.py \
  --pretrained_model_name_or_path=$MODEL_NAME  \
  --instance_data_dir=$INSTANCE_DIR \
  --output_dir=$OUTPUT_DIR \
  --mixed_precision="bf16" \
  --instance_prompt="a photo of sks dog" \
  --resolution=1024 \
  --train_batch_size=1 \
  --guidance_scale=1 \
  --gradient_accumulation_steps=4 \
  --optimizer="prodigy" \
  --learning_rate=1. \
  --report_to="wandb" \
  --lr_scheduler="constant" \
  --lr_warmup_steps=0 \
  --max_train_steps=500 \
  --validation_prompt="A photo of sks dog in a bucket" \
  --validation_epochs=25 \
  --seed="0" \
  --push_to_hub
```

as in [README_flux.md](https://github.com/huggingface/diffusers/blob/main/examples/dreambooth/README_flux.md)

**Comment by sayakpaul** [comment]
Can you try:

```bash
--gradient_checkpointing
--use_8bit_adam
```

**Comment by hlky** [comment]
Could be validation as in #10145

**Comment by huangjun12** [comment]
> Can you try:
> 
> ```shell
> --gradient_checkpointing
> --use_8bit_adam
> ```

I tried this, and turn off validation, it still OOM...

**Comment by sayakpaul** [comment]
@linoytsaban can you give this a look?

**Comment by huangjun12** [comment]
> Can you try:
> 
> ```shell
> --gradient_checkpointing
> --use_8bit_adam
> ```

Have you successfully fine-tuned the full `flux-dev` model using 80GB GPU memory? What are the environment dependencies?

**Comment by timmerscher** [comment]
I'm encountering the same issue!
While tracking it with htop the memory-usage rises above 100GB
![Screenshot from 2024-12-17 15-48-30](https://github.com/user-attachments/assets/72fba63c-a3f7-4f50-b63f-b30a2dede943)

**Comment by Becomebright** [comment]
Any update?

**Comment by liiiiiiiiil** [comment]
using deepspeed，refer to [https://github.com/huggingface/diffusers/blob/main/examples/flux-control/train_control_flux.py](url)

**Comment by joeyism** [comment]
any updates on this?

**Comment by sayakpaul** [comment]
@joeyism what training command have you tried?

**Comment by joeyism** [comment]
@sayakpaul different params for
```bash
accelerate
launch
examples/dreambooth/train_dreambooth_flux.py
--mixed_precision=bf16
--pretrained_model_name_or_path={MODEL_DIR}
--instance_data_dir={img_path}
--output_dir={MODEL_DIR}
--instance_prompt={prompt}
--resolution={config.resolution}
--train_batch_size={config.train_batch_size}
--gradient_accumulation_steps={config.gradient_accumulation_steps}
--learning_rate={config.learning_rate}
--lr_scheduler={config.lr_scheduler}
--lr_warmup_steps={config.lr_warmup_steps}
--max_train_steps={config.max_train_steps}
--checkpointing_steps={config.checkpointing_steps}
--seed={config.seed}
--optimizer=adamW
--use_8bit_adam
--gradient_checkpointing
```

**Comment by sayakpaul** [comment]
Will take a look.

**Comment by sayakpaul** [comment]
```bash
export MODEL_NAME="black-forest-labs/FLUX.1-dev"
export INSTANCE_DIR="dog"
export OUTPUT_DIR="trained-flux"

accelerate launch train_dreambooth_flux.py \
  --pretrained_model_name_or_path=$MODEL_NAME  \
  --instance_data_dir=$INSTANCE_DIR \
  --output_dir=$OUTPUT_DIR \
  --mixed_precision="bf16" \
  --instance_prompt="a photo of sks dog" \
  --resolution=1024 \
  --train_batch_size=1 \
  --guidance_scale=1 \
  --gradient_accumulation_steps=4 \
  --gradient_checkpointing \
  --use_8bit_adam \
  --learning_rate=1e-4 \
  --report_to="wandb" \
  --lr_scheduler="constant" \
  --lr_warmup_steps=0 \
  --max_train_steps=500 \
  --validation_prompt="A photo of sks dog in a bucket" \
  --validation_epochs=25 \
  --seed="0" 
```

I tried with this on a single H100 and even then it failed. @linoytsaban maybe we want to incorporate latent caching in this script too?

**Comment by sayakpaul** [comment]
Cc: @linoytsaban

**Comment by CesureX** [comment]
Has this problem been solved now, and has anyone successfully implemented it?
